### ISSUE TITLE
More ECS task definition EFS volume configuration options

### ISSUE THREAD

**Issue by kabo** [terraform-providers/terraform-provider-aws#12809]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Currently, in [ecs_task_definition.volume.efs_volume_configuration](https://www.terraform.io/docs/providers/aws/r/ecs_task_definition.html#efs-volume-configuration-arguments) the follow options are supported:
- file_system_id
- root_directory

The following options are defined in the [AWS docs](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/efs-volumes.html#specify-efs-config) but do not appear to be supported:
- transitEncryption
- transitEncryptionPort
- authorizationConfig
  - accessPointId
  - iam

### New or Affected Resource(s)

* aws_ecs_task_definition

### Potential Terraform Configuration

```hcl
resource "aws_ecs_task_definition" "service" {
  family                = "service"
  container_definitions = "${file("task-definitions/service.json")}"

  volume {
    name = "service-storage"

    efs_volume_configuration {
      file_system_id = aws_efs_file_system.fs.id
      root_directory = "/opt/data"
      transit_encryption = true
    }
  }
}

```

**Comment by Jukie** [comment]
I'll take a stab at this.

**Comment by jbvmio** [comment]
Any luck @Jukie ?

**Comment by Jukie** [comment]
Yeah I'll be able to push my stuff this weekend. Was waiting for #12715 to get merged and this fell off my plate.

**Comment by aberezin** [comment]
This ticket asks for the following:
1) transitEncryption & transitEncryptionPort
This looks done by @Jukie and is also discussed in 12810 (closed as a dup) 
and is handled by this PR https://github.com/terraform-providers/terraform-provider-aws/pull/13136

2) authorizationConfig and accessPointId
Not sure what happened to this.  I didnt see it in a cursory inspection of @DrFaust92 's PR, whose branch is [here](https://github.com/DrFaust92/terraform-provider-aws/tree/efs-access-point) 

Is anyone working on (2) ?

**Comment by Jukie** [comment]
I have code ready for 2 but like I said it’s dependent on #12715. This is because in order to properly test we need the ability to manage access points with terraform. I validated by creating one manually but without #12715 I can’t build an acceptance test.

**Comment by Jukie** [comment]
The fargate expansion should probably be discussed in a new issue.

**Comment by cgmckeever** [comment]
Has there been any update on this? Interested in the transit encryption .. @Jukie ? Thanks!!

**Comment by Jukie** [comment]
@cgmckeever https://github.com/terraform-providers/terraform-provider-aws/pull/13136

**Comment by rahul824** [comment]
Hi team, 

I have two accesspoints configured in EFS and need to be mounted as separate volumes. I have below config under ecs task definitions. It does not work .. As i understood there is no support for volume configuration for access point till now , how do we approach such cases from terraform side ? I see there is a resource component available for access point , curious  how the access points resources are utilized. 

volume {
            name = "efs-apps"
            efs_volume_configuration {
                file_system_id = <EFS_file_system_ID>
                root_directory = "/accesspoint_1"
            }

          }

**Comment by ckreiling** [comment]
Hi everyone, I'm still fairly new to Terraform and, like @rahul824, am wondering if anyone has found a temporary workaround for this until https://github.com/terraform-providers/terraform-provider-aws/pull/13136 and https://github.com/terraform-providers/terraform-provider-aws/pull/13136 are merged?

**Comment by bflad** [comment]
Support for this functionality has been merged and will release with version 2.68.0 of the Terraform AWS Provider, later this week. Thanks to @Jukie and others for the implementation. 👍
